crates/net/network/src/transactions/mod.rs

```diff
@@ -703,10 +701,8 @@ where
         BroadcastedTransaction: SignedTransaction,
         PooledTransaction: SignedTransaction,
     >,
-    Pool::Transaction: PoolTransaction<
-        Consensus = N::BroadcastedTransaction,
-        Pooled: Into<N::PooledTransaction> + From<RecoveredTx<N::PooledTransaction>>,
-    >,
+    Pool::Transaction:
+        PoolTransaction<Consensus = N::BroadcastedTransaction, Pooled = N::PooledTransaction>,
```

Comment:
hmm, I think this would work for both OP and eth, 
but for OP's case this would mean `Consensus` must be a different type that what we currently have for `TransactionSigned` (minus the deposit variant)

so `Conensus` really just means: All consensus variants that are allowed over p2p

Reply:
couldn't `Consensus` be same as `Pooled` for OP?

Reply:
yes, for op these are the same type, basically `TransactionSigned` - Deposit